miner/worker.go

```diff
@@ -1000,52 +1037,118 @@ func (w *worker) commitNewWork(interrupt *int32, noempty bool, timestamp int64)
 		}
 	}
 	if len(localTxs) > 0 {
-		txs := types.NewTransactionsByPriceAndNonce(w.current.signer, localTxs, header.BaseFee)
-		if w.commitTransactions(txs, w.coinbase, interrupt) {
-			return
+		txs := types.NewTransactionsByPriceAndNonce(env.signer, localTxs, env.header.BaseFee)
+		if w.commitTransactions(env, txs, w.coinbase, interrupt) {
```

Comment:
You should hold a read lock on `worker.mu` here to access `worker.coinbase`.  Maybe the coinbase can just be added as a field in `environment` and provided to `makeEnv` from `prepareWork`.

Reply:
Right! Thanks for catching it.